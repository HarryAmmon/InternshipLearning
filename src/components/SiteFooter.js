import React from 'react';
import { faGithub, faTwitter,} from '@fortawesome/free-brands-svg-icons'
import Copyright from './Copyright.js';
import Styles from './SiteFooter.module.css';
import SocialButton from './SocialButton.js';

const date= new Date();

export default function Footer(){
    return(
        <footer className={Styles.root}>
            <hr className={Styles.line}/>
            <div className={Styles.footerContent}>
                <Copyright year={date.getFullYear()} />
                <div className={Styles.socialMedia}>
                    <SocialButton url="https://github.com/HarryAmmon" name={"GitHub"} />
                    <SocialButton url="https://twitter.com/harry_a99" name={"Twitter"} />
                    <SocialButton url="https://linkedin.com/in/harry-ammon-3a1995150" name={"LinkedIn"} />
                </div>
            </div>
        </footer>
    );
}