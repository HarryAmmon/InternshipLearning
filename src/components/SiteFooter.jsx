import React from "react";
import { faGithub, faTwitter } from "@fortawesome/free-brands-svg-icons";
import Copyright from "./Copyright.jsx";
import Styles from "./SiteFooter.module.scss";
import SocialButton from "./SocialButton.jsx";
import { Hr } from "./Typography/Typography.jsx";

export default function Footer({ date = new Date() }) {
  return (
    <footer className={Styles.root}>
      <Hr />
      <div className={Styles.footerContent}>
        <Copyright year={date.getFullYear()} />
        <div className={Styles.socialMedia}>
          <SocialButton url="https://github.com/HarryAmmon" name={"GitHub"} />
          <SocialButton url="https://twitter.com/harry_a99" name={"Twitter"} />
          <SocialButton
            url="https://linkedin.com/in/harry-ammon-3a1995150"
            name={"LinkedIn"}
          />
        </div>
      </div>
    </footer>
  );
}
