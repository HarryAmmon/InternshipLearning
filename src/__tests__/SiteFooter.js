import React from 'react';
import SiteFooter from "../components/SiteFooter";
import renderer from "react-test-renderer";

describe("The Site footer", () => {
    it("renderers as expected",() => {
        const tree = renderer
            .create(
                <SiteFooter />
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    })
})    expect(tree).toMatchInlineSnapshot(`
      <footer
        className="root"
      >
        <hr
          className="line"
        />
        <div
          className="footerContent"
        >
          <small>
            Copyright Â© 
            2000
             Harry Ammon
          </small>
          <div
            className="socialMedia"
          >
            <a
              className="root"
              href="https://github.com/HarryAmmon"
              target="_blank"
            >
              GitHub
            </a>
            <a
              className="root"
              href="https://twitter.com/harry_a99"
              target="_blank"
            >
              Twitter
            </a>
            <a
              className="root"
              href="https://linkedin.com/in/harry-ammon-3a1995150"
              target="_blank"
            >
              LinkedIn
            </a>
          </div>
        </div>
      </footer>
    `);
  });
});
