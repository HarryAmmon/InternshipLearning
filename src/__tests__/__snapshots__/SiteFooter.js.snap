// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Site footer renderers as expected 1`] = `
<footer
  className="root"
>
  <hr
    className="line"
  />
  <div
    className="footerContent"
  >
    <small>
      Copyright Â© 
      2022
       Harry Ammon
    </small>
    <div
      className="socialMedia"
    >
      <a
        className="root"
        href="https://github.com/HarryAmmon"
        target="_blank"
      >
        GitHub
      </a>
      <a
        className="root"
        href="https://twitter.com/harry_a99"
        target="_blank"
      >
        Twitter
      </a>
      <a
        className="root"
        href="https://linkedin.com/in/harry-ammon-3a1995150"
        target="_blank"
      >
        LinkedIn
      </a>
    </div>
  </div>
</footer>
`;
